core/src/main/scala/kafka/server/ClientRequestQuotaManager.scala

```diff
@@ -46,12 +46,15 @@ class ClientRequestQuotaManager(private val config: ClientQuotaManagerConfig,
   private val exemptMetricName = metrics.metricName("exempt-request-time",
     QuotaType.Request.toString, "Tracking exempt-request-time utilization percentage")
 
-  lazy val exemptSensor: Sensor = getOrCreateSensor(ClientRequestQuotaManager.ExemptSensorName, exemptMetricName)
-
   def recordExempt(value: Double): Unit = {
+    val exemptSensor = getOrCreateExemptSensor()
```

Comment:
The getOrCreateExemptSensor is not a cheap operation, which involves taking the lock among other things. Thus doing it here may slow down the IO threads significantly.
Can we simply remove the lazy keyword for the exemptSensor variable in the original code?

Reply:
I thought about this, but the sensor has an expiration. If we create it during `ClientRequestQuotaManager` initiation, it may not be used (calling `record()`) until request quota is enabled and the sensor could already be expired. Based on the current implementation, `record()` should not be affected by expiration, but `add()` is affected though this method is not used by exemptSensor currently. And once expired, the sensor would be removed from `Metrics.sensors`, which may introduce further issues. On the other hand, `getOrCreateExemptSensor` only acquires a read lock if the sensor exists in Metrics.sensors`. And we've already calling `ClientQuotaManager. getOrCreateQuotaSensors()` each time when sensor is used. `getOrCreateExemptSensor` is similar workload as `getOrCreateQuotaSensors`.